README.md

```diff
@@ -90,192 +97,113 @@ let y = Named["string name"];
-the [API](#API) section.
+  - An _EnumDeclaration_ may have an `extends` clause whose value must be one of %Number%, 
+    %String%, or %Symbol%, which correspond to a <var>hint</var> of `"number"`, `"string"`,
+    or `"symbol"`, respectively. 
+  - If an _EnumDeclaration_ does not have an `extends` clause, it has a _hint_ of `"number"`. 
+  - The _enum members_ of the declaration are evaluated with the provided <var>hint</var>.
+  - An _EnumDeclaration_ may be decorated, and the decorators may add, remove, or modify _enum members_.
+  - An _enum object_ is an Object with a \[\[Prototype]] of `null`.
+  - An _enum object_ has an \[\[EnumMembers]] internal slot, which is a List of the names of its
+    [Enum Members](#enum-members).
+  - An _enum object_ has an @@parseEnum property whose value is a Function that returns the value
+    of the _enum member_ whose name corresponds to the provided argument. 
+    - This member is \[\[Writable]]: `false`, \[\[Configurable]]: `true`, and 
+      \[\[Enumerable]]: `false`.
+  - An _enum object_ has an @@formatEnum property whose value is a Function that returns the name
+    of the first _enum member_ whose value corresponds to the provided argument.
+    - This member is \[\[Writable]]: `false`, \[\[Configurable]]: `true`, and 
+      \[\[Enumerable]]: `false`.
+  - An _enum object_ has an @@toStringTag property whose value is `"Enum"`.
+    - This member is \[\[Writable]]: `false`, \[\[Configurable]]: `true`, and 
+      \[\[Enumerable]]: `false`.
+  - An _enum object_ has an @@iterator property whose value is a Function that returns an iterator
+    for this enum's \[\[EnumMembers]] internal slot where each yielded value is a two-element 
+    array containing the enum member name at index 0 and the enum member value at index 1.
+    - This member is \[\[Writable]]: `false`, \[\[Configurable]]: `true`, and 
+      \[\[Enumerable]]: `false`.
+  - An _enum object_ has a property for each _enum member_ with that _enum member_'s name,
+    whose value is that _enum member_'s value. 
+    - These properties are \[\[Writable]]: `false`, \[\[Configurable]]: `false`, and 
+      \[\[Enumerable]]: `true`.
+  - An _enum object_'s \[\[Extensible]] internal slot is `false`.
 
 ## Enum Members
 
 Enum members consist of a comma-separated list of enum member names with optional initializers:
 
-- Enum names can be identifiers or string literals.
-- The result of the initializer expression must be either `Number`, `String`, `Symbol`, `Boolean`,
-  or another `enum` value.
-- If the result of the initializer expression is another `enum` value, its _underlying primitive 
-  value_ is used instead.
-- The initializer can refer to other named members that have come before it in the same enclosing 
-  lexical `enum` declaration.
+- _enum members_ are evaluated with a supplied <var>hint</hint>, which must be one of: `"number"`,
```

Comment:
what about "bigint"?

Reply:
Since that's Stage 3, I'll add it. An alternative is @weswigham's [suggestion](https://github.com/rbuckton/proposal-enum/pull/2#discussion_r188130138) to use a symbol.

Reply:
I opted for the alternative, as its more extensible.